Runtime/Highlight/DefaultHighlighter.cs

```diff
@@ -0,0 +1,445 @@
+            }
+            
+            if (meshes.Any())
+            {
+                Mesh previewMesh = new Mesh();
+                previewMesh.CombineMeshes(meshes.ToArray());
+                
+                highlightMeshFilter.mesh = previewMesh;
+            }
+            else
+            {
+                throw new NullReferenceException($"{name} has no valid meshes to be highlighted.");
+            }
+        }
+
+        private IEnumerable<CombineInstance> CollectMeshInformationFromMeshFilter(MeshFilter meshFilter)
+        {
+            List<CombineInstance> combinedInstances = new List<CombineInstance>();
+
+            for (int i = 0; i < meshFilter.sharedMesh.subMeshCount; i++)
+            {
+                CombineInstance combineInstance = new CombineInstance
+                {
+                    subMeshIndex = i,
+                    mesh = meshFilter.sharedMesh,
+                    transform = meshFilter.transform.localToWorldMatrix
+                };
+
+                combinedInstances.Add(combineInstance);
+            }
+
+            return combinedInstances;
+        }
+        
+        private IEnumerable<CombineInstance> CollectMeshInformationFromSkinnedMeshRenderer(SkinnedMeshRenderer skinnedMeshRenderer)
+        {
+            List<CombineInstance> combinedInstances = new List<CombineInstance>();
+
+            for (int i = 0; i < skinnedMeshRenderer.sharedMesh.subMeshCount; i++)
+            {
+                CombineInstance combineInstance = new CombineInstance
+                {
+                    subMeshIndex = i,
+                    mesh = skinnedMeshRenderer.sharedMesh,
+                    transform = skinnedMeshRenderer.transform.localToWorldMatrix
+                };
+
+                combinedInstances.Add(combineInstance);
+            }
+
+            return combinedInstances;
+        }
+
+        protected void DisableRenders()
+        {
+            if (highlightMeshRenderer != null)
+            {
+                highlightMeshRenderer.enabled = true;
+                highlightMeshRenderer.gameObject.SetActive(true);
+            }
+
+            foreach (Renderer renderer in renderers)
+            {
+                if (renderer != null)
+                {
+                    renderer.enabled = false;
+                }
+            }
+        }
+
+        protected void ReenableRenderers()
+        {
+            if (highlightMeshRenderer != null)
+            {
+                highlightMeshRenderer.enabled = false;
+                highlightMeshRenderer.gameObject.SetActive(false);
+            }
+
+            foreach (Renderer renderer in renderers)
+            {
+                if (renderer != null)
+                {
+                    renderer.enabled = true;
+                }
+            }
+        }
+
+        protected Material NewHighlightMaterial(Color highlightColor)
+        {
+            Material material = CreateHighlightMaterial();
```

Comment:
should create the material with the shader from GetShader

Reply:
Same as above